deltachat-ios/Controller/WelcomeViewController.swift

```diff
@@ -14,21 +14,22 @@ class WelcomeViewController: UIViewController, ProgressAlertHandler {
 
     private lazy var welcomeView: WelcomeContentView = {
         let view = WelcomeContentView()
-        view.onLogin = { [unowned self] in
-            let accountSetupController = AccountSetupController(dcContext: self.dcContext, editView: false)
+        view.onLogin = { [weak self] in
+            guard let strongSelf = self else { return }
+            let accountSetupController = AccountSetupController(dcContext: strongSelf.dcContext, editView: false)
             accountSetupController.onLoginSuccess = {
-                [unowned self] in
+                [weak self] in
                 if let appDelegate = UIApplication.shared.delegate as? AppDelegate {
                     appDelegate.appCoordinator.presentTabBarController()
                 }
             }
-            self.navigationController?.pushViewController(accountSetupController, animated: true)
+            strongSelf.navigationController?.pushViewController(accountSetupController, animated: true)
         }
-        view.onScanQRCode  = { [unowned self] in
+        view.onScanQRCode  = { [weak self] in
             let qrReader = QrCodeReaderController()
             qrReader.delegate = self
-            self.qrCodeReader = qrReader
-            self.navigationController?.pushViewController(qrReader, animated: true)
+            self?.qrCodeReader = qrReader
```

Comment:
we can use `guard let self = self else { return }` as well. it is very unlikely that between the first and the second `self?` the reference disappears but still it might make sense to have a common pattern for cases where more than once `self?` is called

Reply:
but even when it disappears, things would be fine. in both cases the QrCodeReaderController would not be pushed.

most usages of [weak self] are one-liners, and i would vote for `self?` where possible to keep things short and readable. this is also the way signal uses (also the stringSelf comes from there)

Reply:
maybe exact this occourence was not the best example. but there are others in the PR where popping a view controller or calling a callback would not happen while other logic was already happening.

Reply:
I also wouldn't do it for oneliners

Reply:
i agree, it is better to use the guard if self? would be used multiple times, this avoid thinking about what will happen if self disappears in between, you're right.